Sources/TuistKit/Services/RunService.swift

```diff
@@ -0,0 +1,134 @@
+        targetBuilder: TargetBuilding = TargetBuilder(),
+        targetRunner: TargetRunning = TargetRunner()
+    ) {
+        self.generator = generator
+        self.buildGraphInspector = buildGraphInspector
+        self.targetBuilder = targetBuilder
+        self.targetRunner = targetRunner
+    }
+
+    func run(
+        path: String?,
+        schemeName: String,
+        generate: Bool,
+        clean: Bool,
+        configuration: String?,
+        device: String?,
+        version: String?,
+        arguments: [String]
+    ) throws {
+        let runPath: AbsolutePath
+        if let path = path {
+            runPath = AbsolutePath(path, relativeTo: FileHandler.shared.currentPath)
+        } else {
+            runPath = FileHandler.shared.currentPath
+        }
+
+        let graph: ValueGraph
+        if try (generate || buildGraphInspector.workspacePath(directory: runPath) == nil) {
+            logger.notice("Generating project for running", metadata: .section)
+            graph = try generator.generateWithGraph(path: runPath, projectOnly: false).1
+        } else {
+            graph = try generator.load(path: runPath)
+        }
+        let graphTraverser = ValueGraphTraverser(graph: graph)
+        let runnableSchemes = buildGraphInspector.runnableSchemes(graphTraverser: graphTraverser)
+
+        logger.debug("Found the following runnable schemes: \(runnableSchemes.map(\.name).joined(separator: ", "))")
+
+        guard let scheme = runnableSchemes.first(where: { $0.name == schemeName }) else {
+            throw RunServiceError.schemeNotFound(scheme: schemeName, existing: runnableSchemes.map(\.name))
+        }
+
+        guard let graphTarget = buildGraphInspector.runnableTarget(scheme: scheme, graphTraverser: graphTraverser) else {
+            throw RunServiceError.schemeWithoutRunnableTarget(scheme: scheme.name)
+        }
+
+        try targetRunner.assertCanRunTarget(graphTarget.target)
+
+        try targetBuilder.buildTarget(
+            graphTarget,
+            schemeName: scheme.name,
+            clean: clean,
+            configuration: configuration,
+            buildOutputPath: nil
+        )
+
+        let minVersion: Version?
+        if let deploymentTarget = graphTarget.target.deploymentTarget {
+            minVersion = deploymentTarget.version.version()
+        } else {
+            minVersion = scheme.targetDependencies()
+                .flatMap {
+                    graphTraverser
+                        .directLocalTargetDependencies(path: $0.projectPath, name: $0.name)
+                        .map(\.target)
+                        .map(\.deploymentTarget)
+                        .compactMap { $0?.version.version() }
+                }
+                .sorted()
+                .first
+        }
+
+        let version: Version? = try version.map { versionString in
+            guard let version = Version(string: versionString) else {
```

Comment:
@luispadron bear in mind that this class doesn't take `14.5` as valid because it expects a string that adheres to semantic versioning. Because of that the following command fails:

```
swift run tuist run MyApp --path ~/Downloads/MyApp --device "iPhone 12" --os "iOS 14.5"
```

Reply:
You're right! I fixed this up by using `.version()` which `tuist test` uses as well. This takes into account the scenario you showcased.